lib/msf/core/post/windows/task_scheduler.rb

```diff
@@ -0,0 +1,226 @@
+              OptEnum.new(
+                'ScheduleType', [
+                  true,
+                  'Schedule frequency for the new created task.',
+                  'ONSTART',
+                  %w[MINUTE HOURLY DAILY WEEKLY MONTHLY ONCE ONSTART ONLOGON ONIDLE]
+                ]
+              ),
+              OptInt.new(
+                'ScheduleModifier', [
+                  false,
+                  'Schedule frequency modifier. This defines the amount of minutes/hours/days/weeks/months,'\
+                  ' depending on the ScheduleType value. When ONIDLE type is used, this represents how many'\
+                  ' minutes the computer is idle before the task starts. This value is not used with ONCE, '\
+                  'ONSTART and ONLOGON types.',
+                  1
+                ]
+              ),
+              OptString.new(
+                'ScheduleRemoteSystem', [
+                  false,
+                  'The remote system to connect to. Use a FQDN to avoid long delays due to automatic DNS resolutions.'
+                ]
+              ),
+              OptString.new(
+                'ScheduleUsername', [
+                  false,
+                  'Use the permissions of this remote user account to schedule the task remotely (e.g. '\
+                  'MYDOMAIN\User1 or .\Administrator). This requires \'ScheduleRemoteSystem\' to be set.'
+                ]
+              ),
+              OptString.new(
+                'SchedulePassword', [
+                  false,
+                  'Password of the remote user account set in \'ScheduleUsername\'. This requires '\
+                  '\'ScheduleUsername\' and \'ScheduleRemoteSystem\' to be set.'
+                ]
+              ),
+              OptString.new(
+                'ScheduleRunAs', [
+                  false,
+                  'Execute the task under this user account (default: SYSTEM).',
+                  'SYSTEM'
+                ]
+              ),
+              OptBool.new(
+                'ObfuscateTask', [
+                  false,
+                  'Hide the task from "schtasks /query" and Task Scheduler by deleting the '\
+                  'associated Security Descriptor registry value. Note that SYSTEM privileges '\
+                  'are needed for this. It will try to elevate privileges if the session is not '\
+                  'already running under the SYSTEM user.',
+                  true
+                ]
+              )
+            ], self.class
+          )
+        end
+
+        def get_schtasks_cmd_string(schtasks_cmd, opts = {})
+          schtasks_cmd.prepend('schtasks')
+          system = opts[:remote_system] || datastore['ScheduleRemoteSystem']
+          schtasks_cmd += ['/s', system] if system
+          username = opts[:username] || datastore['ScheduleUsername']
+          schtasks_cmd += ['/u', username] if username
+          password = opts[:password] || datastore['SchedulePassword']
```

Comment:
> Check if it possible to use the current user credentials instead of specifying them here.

Yes, it is possible and the code has been adjusted to support this 🎉 

> Also, check if schtasks option are consistent across Windows versions.

The code has also been updated to support older Windows versions